File: ninja/python/indovina.py
import random

def gioca():
    segreto = random.randint(1, 50)
    tentativi = 0
    print("=== INDOVINA IL NUMERO ===")

    while True:
        scelta = int(input("Indovina il numero tra 1 e 50: "))
        tentativi += 1

        if scelta < segreto:
            print("Troppo basso! Riprova.")
        elif scelta > segreto:
            print("Troppo alto! Riprova.")
        else:
            print(f"Complimenti! Hai indovinato il numero {segreto} in {tentativi} tentativi.")
            break

if __name__ == "__main__":
    gioca()

File: ninja/c++/indovina.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

int main() {
    std::srand(std::time(0));

    int NumeroSegreto = std::rand() % 100 +1;
    int tentativo = 0;
    int tentativifatti = 0;
    std::cout << "Indovina il numero tra 1 e 100!\n" << std::endl;

    do {
        std::cout << "Inserisci il numero: ";
        std::cin >> tentativo;
        tentativifatti++;

        if (tentativo < NumeroSegreto) {
            std::cout << "Troppo basso! Riprova.\n" << std::endl;
        } else if (tentativo > NumeroSegreto) {
            std::cout << "Troppo alto! Riprova.\n" << std::endl;
        } else {
            std::cout << "Congratulazioni! Hai indovinato il numero " << NumeroSegreto << " in " << tentativifatti << " tentativi.\n" << std::endl;
        }
    } while (tentativo != NumeroSegreto);
}
